### ISSUE TITLE
Linux (Wayland) support 🌱

### ISSUE THREAD

**Issue by shivanshs9** [orhun/menyoki#2]
**Is your feature request related to a problem? Please describe.**
I love the concept and would like to give it a try, but my Linux system is wayland-based :disappointed:. Hope wayland support is added soon :pray: 

**Describe the solution you'd like**
There are existing tools for wlroots-based compositors, like https://github.com/ammen99/wf-recorder and https://github.com/emersion/grim. Hope it helps you to figure out the implementation support for Wayland. I'll be glad to help out too if it's needed.

**Comment by orhun** [comment]
Hey!
Hopefully I'll be able to [implement](https://github.com/orhun/menyoki/blob/master/IMPLEMENTATION.md) **menyoki** for Wayland _soon_. Although it looks like there are some obstacles which I stumbled upon while checking out the similar projects: https://github.com/phw/peek#why-no-native-wayland-support

It's good to have a pinned issue about this though. Thanks :)

**Comment by shivanshs9** [comment]
Indeed, it's impossible to get window information and coordinates in Wayland due to security reasons. However, instead of window, if menyoki uses screen coordinates, then that information can be easily obtained, like in https://github.com/emersion/slurp tool. It's even possible to get coordinates of the specified window using slurp in sway compositor only.

As for screenshot capability, I believe wlroots-based compositor have a protocol for screencopy, albeit unstable - https://github.com/swaywm/wlroots/blob/master/protocol/wlr-screencopy-unstable-v1.xml
The same protocol is being used by grim tool.

**Comment by AkechiShiro** [comment]
Any news on this ? How can the community help move this issue forward ?

**Comment by jokeyrhyme** [comment]
There's also XDG desktop portals, which abstract application developers from needing to know the specifics of compositors: https://flatpak.github.io/xdg-desktop-portal/#gdbus-org.freedesktop.portal.ScreenCast

**Comment by orhun** [comment]
> Any news on this? How can the community help move this issue forward?

I don't have a Wayland system to test this out so I can't really move forward with the implementation. I think someone who uses Wayland can just jump right into it and implement the [missing methods](https://github.com/orhun/menyoki/blob/master/IMPLEMENTATION.md#implementing-for-other-platforms). After that, I believe `menyoki` should just work out of the box.

**Comment by orhun** [comment]
One thing to note here, I'm pretty much okay with using 3rd party capture tools for Wayland (AKA shelling out from `menyoki`) if native capture support in Wayland is problematic.

**Comment by orhun** [comment]
Now implemented: #57
